### PR TITLE
fixed evaluation of javascript UDF

### PR DESCRIPTION
**Pull request by chaudum**
Providing a JavaScript UDF with EcmaScript 6 arrow function syntax, such as
`var f = (x) => x;`, caused CrateDB to crash, because Java does not
throw a ScriptException but an AssertionError (see
https://bugs.openjdk.java.net/browse/JDK-8144711).

### CODE REVIEW

File: enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java [hunk 1]
```diff
@@ -116,7 +117,24 @@ public void testInvalidJavascript() throws ScriptException{
         );
 
         String validation = udfService.getLanguage(JS).validate(udfMeta);
-        assertThat(validation, startsWith("Invalid JavaScript in function 'doc.f(double)': <eval>:1:27"));
+        assertThat(validation, startsWith("Invalid JavaScript in function 'doc.f(double)'"));
+        assertThat(validation, endsWith("^ in <eval> at line number 1 at column number 27"));
+    }
+
+    @Test
+    public void testInvalidJavascriptThrowsAssertion() {
```
Comment (mxm) [inline enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java]: Would make this `testInvalidJavascriptThrowsAssertion*Error*() {` because I was confused at first how it can throw an assertion.

Comment (chaudum) [reply enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java]: Yes, you're right of course, it is a `java.lang.AssertionError`.

File: enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java [hunk 2]
```diff
@@ -116,7 +117,24 @@ public void testInvalidJavascript() throws ScriptException{
         );
 
         String validation = udfService.getLanguage(JS).validate(udfMeta);
-        assertThat(validation, startsWith("Invalid JavaScript in function 'doc.f(double)': <eval>:1:27"));
+        assertThat(validation, startsWith("Invalid JavaScript in function 'doc.f(double)'"));
+        assertThat(validation, endsWith("^ in <eval> at line number 1 at column number 27"));
+    }
+
+    @Test
+    public void testInvalidJavascriptThrowsAssertionError() {
```
Comment (mfussenegger) [inline enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java]: If I may join in on the naming discussion, I think something along the lines of `test_validate_catches_assertion_errors` or `test_validate_never_raises` would be more accurate.

File: CHANGES.txt [hunk 3]
```diff
@@ -21,6 +21,10 @@ Changes
 Fixes
 =====
 
+- Fixed the evaluation of JavaScript user defined functions that caused CrateDB
+  to crash because of an unhandled assertion when providing the UDF with
+  EcmaScript 6 arrow function syntax (``var f = (x) => x;``).
```
Comment (mfussenegger) [inline CHANGES.txt]: Do you think it's worth mentioning that this only was the case under certain JVM versions?

File: enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java (hunk 2 above)
Comment (mxm) [reply enterprise/lang-js/src/test/java/io/crate/operation/language/JavascriptUserDefinedFunctionTest.java]: 👍 

Would rephrase to `testJavascriptExceptionsAreCaught`

File: CHANGES.txt (hunk 3 above)
Comment (chaudum) [reply CHANGES.txt]: The problem occurs both with OpenJDK and OracleJDK and is only fixed in Java 9. Since we do not support Java 9 yet, I think it is not necessary to mention anything about versions.
